test/controllers/workshops_controller_test.rb

```diff
@@ -84,4 +89,24 @@ class WorkshopsControllerTest < ActionDispatch::IntegrationTest
       get '/personal_website/FAKE.md'
     end
   end
+
+  test "the proper page event is sent when workshop roots are loaded" do
+    # Sign in
+    get '/auth/github/callback'
+
+    # Load a workshop
+    get '/personal_website/'
```

Comment:
Why the extra comments? The get for sign in is confusing, but it's used everywhere else in the tests so I don't think it deserves its own comments.